src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/delay/impl/RetryConsumerDelayStrategy.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.retry.delay.impl;
+
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.retry.delay.ConsumerDelayStrategy;
+
+import java.util.HashMap;
+
+public class RetryConsumerDelayStrategy implements ConsumerDelayStrategy {
+
+    private final ApplicationLogger logger;
+
+    @Value("${RETRY_THROTTLE_RATE_SECONDS}")
+    private long retryThrottleRateSeconds;
+
+    public RetryConsumerDelayStrategy(ApplicationLogger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void throttle() {
+        try {
+            logger.info("Delaying retry consumer for " + retryThrottleRateSeconds + " seconds");
+            long retryThrottleRateMilliseconds = retryThrottleRateSeconds * 1000;
+            Thread.sleep(retryThrottleRateMilliseconds);
+            logger.info("Delay period over");
+        } catch(InterruptedException ie) {
+            logger.error("Error putting retry thread to sleep", ie, new HashMap<>());
+            Thread.currentThread().interrupt();
```

Comment:
In this scenario most likely case is app is attempting to shut down.
So mark the thread as interrupted and then return from the method to allow the application to proceed quickly.

Reply:
Not sure what I would change here interrupt is called and it falls out of catch.

Reply:
Sorry I wasn't clear, 
I meant explicitly adding a `return;`
Appreciate the behaviour is identical, but that might change down the line (I doubt it but can't be too safe)

Reply:
OK